ee/tabby-webserver/src/repositories/repo.rs

```diff
@@ -0,0 +1,122 @@
+use std::{collections::HashMap, path::PathBuf, str::FromStr};
+
+use anyhow::Result;
+use axum::{
+    body::boxed,
+    http::{header, Request, Uri},
+    response::{IntoResponse, Response},
+    Json,
+};
+use hyper::Body;
+use serde::{Deserialize, Serialize};
+use tabby_common::{to_filename, SourceFile};
+use tokio::sync::OnceCell;
+use tower::ServiceExt;
+use tower_http::services::ServeDir;
+use tracing::error;
+
+pub(crate) static DATASET: OnceCell<HashMap<DatasetKey, SourceFile>> = OnceCell::const_new();
```

Comment:
Instead of exporting a static variable, expose a function with following signature:
```
pub async fn resolve_meta(...) -> Meta
```